content/en/blog/2023/any-metric-receiver.md

```diff
@@ -0,0 +1,480 @@
+              \"value\": {
+                \"stringValue\": \"${URL}\"
+              }
+            }
+          ]
+        },
+        \"scopeMetrics\": [
+          {
+            \"metrics\": [
+              {
+                \"name\": \"tls.server.not_after.time_left\",
+                \"unit\": \"s\",
+                \"description\": \"\",
+                \"gauge\": {
+                  \"dataPoints\": [
+                    {
+                      \"asInt\": ${secondsLeft},
+                      \"timeUnixNano\": ${now}000000000
+                    }
+                  ]
+                }
+              }
+            ]
+          }
+        ]
+      }
+    ]
+  }
+"
+curl -X POST -H "Content-Type: application/json" -d "${data}" -i localhost:4318/v1/metrics
+```
+
+Activate the OTLP Receiver in your `collectors-config`:
+
+```yaml
+receivers:
+  otlp:
+    protocols:
+      http:
+      grpc:
+exporters:
+  debug:
+    verbosity: detailed
+service:
+  pipelines:
+    metrics:
+      receivers: [otlp]
+      exporters: [debug]
+```
+
+Execute your updated `ssl_check.sh`:
+
+```shell
+./ssl_check.sh opentelemetry.io
+```
+
+This time, your metric will be presented with the correct unit set and the
+resource reported as defined in your JSON:
+
+```shell
+2023-11-24T15:28:51.212+0100	info	ResourceMetrics #0
+Resource SchemaURL:
+Resource attributes:
+     -> service.name: Str(opentelemetry.io)
+ScopeMetrics #0
+ScopeMetrics SchemaURL:
+InstrumentationScope
+Metric #0
+Descriptor:
+     -> Name: tls.server.not_after.time_left
+     -> Description:
+     -> Unit: s
+     -> DataType: Gauge
+NumberDataPoints #0
+StartTimestamp: 1970-01-01 00:00:00 +0000 UTC
+Timestamp: 2023-11-24 14:28:51 +0000 UTC
+Value: 4347231
+	{"kind": "exporter", "data_type": "metrics", "name": "debug"}
+```
+
+Working with JSON in shell scripts isn't particularly desirable, as this example
+clearly demonstrates! While there are methods for improvement, you may
+ultimately find it more efficient to use a language such as Python or Node.js,
+or incorporate metrics (with gauge support) into your preferred OTel CLI tool!
+
+## Summary
+
+In this blog post you learned how you can use a _catch-all_ receiver like the
+Carbon Receiver to feed any metric into your OpenTelemetry Collector. This
+approach is recommended, if none of the available receivers meets your needs,
+and if writing your own receiver is to time consuming to accomplish your goal,
+because you are not familiar with Go or you don't have the bandwidth to rewrite
+your existing solution (a shell script) to Go.
+
+As outlook, you learned about the possibility to send your metrics to the
+OpenTelemetry Collector directly using OTLP and `curl`. This approach is
+recommended, if you can not modify the pipelines of your OpenTelemetry
+collector. It will also become a valid alternative to _catch-all_ receivers with
+the availability of command line tools that export metrics via OTLP.
```

Comment:
```suggestion
You learned how to send your metrics to the
OpenTelemetry Collector directly using OTLP and `curl`. Use this approach when you cannot modify the pipelines of your OpenTelemetry
collector. It will also become a valid alternative to _catch-all_ receivers with
the availability of command line tools that export metrics via OTLP.
```